x-pack/plugins/security_solution/public/common/components/exceptions/add_exception_modal/index.test.tsx

```diff
@@ -0,0 +1,339 @@
+          <AddExceptionModal
+            ruleId={'123'}
+            ruleIndices={['filebeat-*']}
+            ruleName={ruleName}
+            exceptionListType={'detection'}
+            onCancel={() => {}}
+            onConfirm={() => {}}
+            alertData={alertDataMock}
+          />
+        </ThemeProvider>
+      );
+      const callProps = ExceptionBuilderComponent.mock.calls[0][0];
+      act(() => callProps.onChange({ exceptionItems: [getExceptionListItemSchemaMock()] }));
+    });
+    it('has the add exception button enabled', () => {
+      expect(
+        wrapper.find('button[data-test-subj="add-exception-confirm-button"]').getDOMNode()
+      ).not.toBeDisabled();
+    });
+    it('should render the exception builder', () => {
+      expect(wrapper.find('[data-test-subj="alert-exception-builder"]').exists()).toBeTruthy();
+    });
+    it('should not prepopulate endpoint items', () => {
+      expect(defaultEndpointItems).not.toHaveBeenCalled();
+    });
+    it('should render the close on add exception checkbox', () => {
+      expect(
+        wrapper.find('[data-test-subj="close-alert-on-add-add-exception-checkbox"]').exists()
+      ).toBeTruthy();
+    });
+    it('should have the bulk close checkbox disabled', () => {
+      expect(
+        wrapper
+          .find('input[data-test-subj="bulk-close-alert-on-add-add-exception-checkbox"]')
+          .getDOMNode()
+      ).toBeDisabled();
+    });
+  });
+
+  describe('when there is bulk-closeable alert data passed to an endpoint list exception', () => {
+    let wrapper: ReactWrapper;
+    let callProps: {
+      onChange: (props: { exceptionItems: ExceptionListItemSchema[] }) => void;
+      exceptionListItems: ExceptionListItemSchema[];
+    };
+    beforeEach(() => {
+      // Mocks the index patterns to contain the pre-populated endpoint fields so that the exception qualifies as bulk closable
+      (useFetchIndexPatterns as jest.Mock).mockImplementation(() => [
+        {
+          isLoading: false,
+          indexPatterns: {
+            ...stubIndexPattern,
+            fields: [
+              { name: 'file.path.text', type: 'string' },
+              { name: 'subject_name', type: 'string' },
+              { name: 'trusted', type: 'string' },
+              { name: 'file.hash.sha256', type: 'string' },
+              { name: 'event.code', type: 'string' },
+            ],
+          },
+        },
+      ]);
+      const alertDataMock: { ecsData: Ecs; nonEcsData: TimelineNonEcsData[] } = {
+        ecsData: { _id: 'test-id' },
+        nonEcsData: [{ field: 'file.path', value: ['test/path'] }],
+      };
+      wrapper = mount(
+        <ThemeProvider theme={() => ({ eui: euiLightVars, darkMode: false })}>
+          <AddExceptionModal
+            ruleId={'123'}
+            ruleIndices={['filebeat-*']}
+            ruleName={ruleName}
+            exceptionListType={'endpoint'}
+            onCancel={() => {}}
```

Comment:
nit: If you search through the code base most of the time people across the plugins and code base do this when stubbing out no operations:

```ts
jest.fn()
```

Makes it easier as a stub that it won't blow up later on as it's more accepting of what can call it, etc...

For example:

```ts
onCancel={jest.fn()}
```